test/src/core/Boot.js

```diff
@@ -194,30 +194,33 @@ describe('Property default comparators', function() {
 
   it('accepts ===', function() {
     var s = '';
-    expect(c(s, s)).toEqual(0);
+    expect(c(s, s)).toBe(0);
   });
   it('accepts double falsey', function() {
-    expect(c(false, false)).toEqual(0);
-    expect(c(NaN, NaN)).toEqual(0);
-    expect(c(0, 0)).toEqual(0);
-    expect(c('', '')).toEqual(0);
+    expect(c(false, false)).toBe(0);
+    expect(c(NaN, NaN)).toBe(0);
+    expect(c(0, 0)).toBe(0);
+    expect(c('', '')).toBe(0);
   });
   it('accepts left falsey', function() {
-    expect(c(false, 6)).toEqual(-1);
-//    expect(c(NaN, 6)).toEqual(-1); // Is this really what we expect?
+    expect(c(false, 6)).toEqual(1);
+    expect(c(NaN, 6)).toEqual(1); // Is this really what we expect?
     expect(c(0, 6)).toEqual(-1);
     expect(c('', 6)).toEqual(-1);
   });
   it('accepts right falsey', function() {
-//    expect(c(4, false)).toEqual(1);
-//    expect(c(4, NaN)).toEqual(1);
+    expect(c(4, false)).toEqual(-1);
+    expect(c(4, NaN)).toEqual(-1);
     expect(c(4, 0)).toEqual(1);
     expect(c(4, '')).toEqual(1);
   });
   it('falls back on $UID.compareTo', function() {
-    var o1 = {};
-    var o2 = {};
-    expect(c(o1, o2)).toEqual(-1);
+    var o1 = Object.create(null);
+    o1.$UID = 26;
+    var o2 = Object.create(null);
+    o2.$UID = 27;
+    expect(c(o1, o2)).toBe(-1);
+    expect(c(o2, o1)).toBe(1);
```

Comment:
There were a few instances on my local machine where `o1` seemed to have a greater UID than `o2`. I have explicitly defined them to avoid this from causing a false test failure